### PR TITLE
feat: add .downloads API

### PR DESCRIPTION
**Pull request by tony-go**
Hey 👋 

This PR resolve issue #22 

## Features
* new `.downloads` API, I restrict the interface to
  * one package at a time (the "all packages" is not available) 
  * three-period only: `last-days` | `last-week` | `last-month`
  -> Please tell me if you have concerns with that, extending the interface should not be okay

## Fixes:
* 🎉 tests work again (we are now using `mocha` and `chai`)

## Notes:
* I had an issue with the `--parallel` mode of mocha but didn't have enough time to debug (I will definitely)
* I had to comment test that previously used `jest.mock`, but I'll raise a PR after that one
* I didn't take a look at CI (But I'll too haha)

### CODE REVIEW

**Comment by fraxken** [comment]
Note for another PR (for tests i guess). You can use Undici Mock API to catch http requests. That what we do in NodeSecure/vuln for example: https://github.com/NodeSecure/vuln/blob/main/test/strategies/unit/sonatype.js#L37-L46

File: src/api.ts [hunk 1]
```diff
@@ -3,9 +3,32 @@ import * as npm from "@npm/types";
 
 // Import Internal Dependencies
 import * as httpie from "@myunisoft/httpie";
-import { httpRegistryAgent, getLocalRegistryURL } from "./registry.js";
+import { httpRegistryAgent, getLocalRegistryURL, getNpmAPIURL } from "./registry.js";
 import { clamp } from "./utils.js";
 
+export type Period = "last-day" | "last-month" | "last-week";
+
+interface NpmPackageDownload {
```
[the remark this answers is not in the document]
Comment (tony-go) [reply src/api.ts:11]: Right! Sure I said to myself but finally forgot.

**Comment by tony-go** [comment]
> Note for another PR (for tests i guess). You can use Undici Mock API to catch http requests. That what we do in NodeSecure/vuln for example: https://github.com/NodeSecure/vuln/blob/main/test/strategies/unit/sonatype.js#L37-L46

HOUUUUUUUUUUU 🔥

**Comment by tony-go** [comment]
Forgot to mention that my main concern here was about DI. I used the third argument to do it, but maybe a factory should be more appropriate?

**Comment by fraxken** [comment]
> Forgot to mention that my main concern here was about DI. I used the third argument to do it, but maybe a factory should be more appropriate?

@tony-go For me that look like test induced degradation. The mock API of undici is the way to go (no need for this third argument).

**Comment by tony-go** [comment]
> @tony-go For me that look like test induced degradation. The mock API of undici is the way to go (no need for this third argument).

Yeah agreed! But also, we should keep in mind that a big part of the logic is about building the URL: I'll export this piece to test unitary and test the main function with Undici Mock API as you suggest.

Sounds good?

**Comment by fraxken** [comment]
@tony-go yep, go 👍

**Review by fraxken — APPROVED** [review]

**Comment by fraxken** [comment]
@all-contributors please add @tony-go for code

**Pull request merged**